src/web/app/src/components/ThemeSelector.tsx

```diff
@@ -0,0 +1,133 @@
+// eslint-disable-next-line no-use-before-define
+import React, { useState } from 'react';
+import {
+  IconButton,
+  Popover,
+  Typography,
+  Paper,
+  Tooltip,
+  Zoom,
+  Divider,
+  MenuList,
+  MenuItem,
+  ListItemText,
+} from '@material-ui/core';
+import PaletteIcon from '@mui/icons-material/Palette';
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import { useTheme } from './ThemeProvider';
+
+const lightDefaultLogoUrl = '/colorThemes/light-default.png';
+const darkDefaultLogoUrl = '/colorThemes/dark-default.png';
+const lightContrastLogoUrl = '/colorThemes/light-high-contrast.png';
+const darkDimLogoUrl = '/colorThemes/dark-dim.png';
+const themes = [
+  { title: 'Light Default', id: 'light-default', image: lightDefaultLogoUrl },
+  { title: 'Light High Contrast', id: 'light-high-contrast', image: lightContrastLogoUrl },
+  { title: 'Dark Default', id: 'dark-default', image: darkDefaultLogoUrl },
+  { title: 'Dark Dim', id: 'dark-dim', image: darkDimLogoUrl },
+];
+
+const useStyles = makeStyles((theme) => ({
+  themeSelectorButton: {
+    color: theme.palette.primary.main,
+    '&:hover': {
+      backgroundColor: 'transparent',
+      color: theme.palette.text.primary,
+    },
+  },
+  popoverPaper: {
+    color: theme.palette.text.primary,
+    backgroundColor: theme.palette.background.paper,
+  },
+  activatedButton: {
+    '&:hover': {
+      color: theme.palette.text.secondary,
+    },
+  },
+}));
+
+const ButtonTooltip = withStyles({
+  tooltip: {
+    fontSize: '1.5rem',
+    margin: 0,
+  },
+})(Tooltip);
+
+const ThemeSelector = () => {
+  const classes = useStyles();
+  const { preferredTheme, changeTheme } = useTheme();
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+
+  const handleClick = (e: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(e.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleChange = (id: String): void => {
+    console.log('theme in ThemeSelector', preferredTheme, id);
+    changeTheme(id);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
```

Comment:
Optionally, you can cast `anchorEl` to a boolean type using the double exclamation operator, like:

```ts
const id = !!anchorEl;
```

Reply:
Yeah, it's clearer. Thanks

Reply:
This one also gives me `Redundant double negation               no-extra-boolean-cast` lint error.

Reply:
[This comment](https://github.com/Seneca-CDOT/telescope/pull/3761/files#r1027118153) also explains what happened here.